<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/home/z/Datasets/tunnels1.bag"/>

  <node name="rovio" pkg="rovio" type="rovio_node" output="screen">
    <param name="filter_config" value="$(find mav_startup)/parameters/sensors/lidarstick/rovio_filter.info" />
    <param name="camera0_config" value="$(find mav_startup)/parameters/sensors/lidarstick/rovio_cam0.yaml" />
    <remap from="cam0/image_raw" to="uvc_camera/cam_1/image_raw"/>
    <remap from="imu0" to="uvc_camera/adis_imu"/>
    <param name="world_frame" value="world"/>
  </node>

  <node name="player" pkg="rosbag" type="play" output="screen" args="-r 1.0 -s 1 --clock $(arg bag_file)" if="$(arg play_bag)">
    <remap from="/tf" to="/not_tf"/>
  </node>

   <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="/velodyne_points"/>
    <param name="tsdf_voxel_size" value="0.05" />
    <param name="truncation_distance" value="0.5"/>
    <param name="color_mode" value="normals" />
    <param name="icp_iterations" value="10"/>
    <param name="icp_subsample_keep_ratio" value="0.2"/>
    <param name="icp_min_match_ratio" value="0.25"/>
    <param name="use_tf_transforms" value="false" />
    <param name="update_mesh_every_n_sec" value="5.0" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="mesh_min_weight" value="2"/>
    <param name="method" value="fast" />
    <param name="max_ray_length_m" value="10.0" />
    <param name="use_const_weight" value="true" />
    <param name="allow_clear" value="true" />
    <param name="verbose" value="true" />
    <remap from="transform" to="/rovio/transform" />
    <rosparam file="$(find voxblox_ros)/cfg/tunnels.yaml"/>
    <param name="mesh_filename" value="/home/z/Desktop/lidar.ply" />
  </node>

</launch>

